optimizely/helpers/audience.py

```diff
@@ -11,35 +11,51 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
+
 from . import condition as condition_helper
 from . import condition_tree_evaluator
+from .enums import AudienceEvaluationLogs as logs
 
 
-def is_user_in_experiment(config, experiment, attributes):
+def is_user_in_experiment(config, experiment, attributes, logger):
   """ Determine for given experiment if user satisfies the audiences for the experiment.
 
   Args:
     config: project_config.ProjectConfig object representing the project.
     experiment: Object representing the experiment.
     attributes: Dict representing user attributes which will be used in determining
                 if the audience conditions are met. If not provided, default to an empty dict.
+    logger: Provides a logger to send log messages to.
 
   Returns:
     Boolean representing if user satisfies audience conditions for any of the audiences or not.
   """
 
   # Return True in case there are no audiences
   audience_conditions = experiment.getAudienceConditionsOrIds()
+
   if audience_conditions is None or audience_conditions == []:
+    logger.info(logs.NO_AUDIENCE_ATTACHED.format(
+      experiment.key
+    ))
+
     return True
 
+  logger.debug(logs.EVALUATING_AUDIENCES.format(
+    experiment.key,
+    json.dumps(audience_conditions)
+  ))
+
+  logger.debug(logs.USER_ATTRIBUTES.format(json.dumps(attributes)))
```

Comment:
The doc sort of suggests that we log attributes with every audience evaluation. I have logged it here at top-level since attributes remain same irrespective of audience conditions